### ISSUE TITLE
Process wikidump with WikiExtractor

### ISSUE THREAD

**Issue by 1024er** [jzhoubu/HLP#2]
Hello, I see that WikiExtractor has some options, do you any of use these options ?

usage: wikiextractor [-h] [-o OUTPUT] [-b n[KMG]] [-c] [--json] [--html] [-l] [-ns ns1,ns2]
			 [--templates TEMPLATES] [--no-templates] [--html-safe HTML_SAFE] [--processes PROCESSES]
			 [-q] [--debug] [-a] [-v]
			 input

Thank you

**Comment by 1024er** [comment]
I used 
python -m wikiextractor.WikiExtractor --no-templates --json --processes 200 enwiki-latest-pages-articles.xml

and got 16 million passages, about 72% amount of your provided "22 million passages".

So will you please share your wikiextractor command ? thank you @jzhoubu

**Comment by jzhoubu** [comment]
Hi @1024er . Unfortunately, I didn't store any of the wikiextractor commands. As far as I can recall, I didn't use any option related to templates.

**Comment by 1024er** [comment]
Thank you for your reply. @jzhoubu 

I also tried :
 python -m wikiextractor.WikiExtractor --json --processes 200 ../enwiki-latest-pages-articles.xml

still got 16 million passages.


---


will you please provide the full name of "snapshot 03-01-2021 of an English Wikipedia dump",  I am not sure whether it is March 01, or January 03. 

Thank you

**Comment by jzhoubu** [comment]
> will you please provide the full name of "snapshot 03-01-2021 of an English Wikipedia dump", I am not sure whether it is March 01, or January 03.
> 
> Thank you

It's March 01. 
Meanwhile, I think either Jan 2021 or March 2021 can exact more than 21m passages. DPR uses an older Wikipedia dump and has 21m passages. This highly depends on the preprocessing pipelines.

**Comment by 1024er** [comment]
One more thing, I finetune the provided ckpt [hlp20210726.best](https://drive.google.com/file/d/10vg8-6S4Rnn7wZOxfBmn2C4il0oV0Oyk/view?usp=sharing) on NQ dataset:

python -m torch.distributed.launch --nproc_per_node=8 train_dense_encoder.py \
    hydra.run.dir=./experiments/finetune_nq/train \
    model_file=../../pretrain_hlp/train/dpr_biencoder.best \
    train_datasets=[nq_train] dev_datasets=[nq_dev] \
    train=pretrain_8xV100

Then I generate the embedding and perform the retrieval evalutaion. But the results are lower than zeroshot, I am confused.

`
[2022-06-12 01:21:41,611][dpr.data.qa_validation][INFO] - Matching answers in top docs...
[2022-06-12 01:21:46,691][dpr.data.qa_validation][INFO] - Per question validation results len=3610
[2022-06-12 01:21:46,708][root][INFO] - Validation results: top k documents hits [835, 1208, 1454, 1608, 1718, 1829, 1917, 1987, 2042, 2096, 2150, 2188, 2215, 2239, 2271, 2302, 2329, 2354, 2370, 2389, 2406, 2423, 2439, 2463, 2479, 2497, 2512, 2520, 2531, 2543, 2552, 2560, 2575, 2582, 2591, 2601, 2614, 2620, 2626, 2631, 2640, 2650, 2655, 2664, 2672, 2680, 2686, 2691, 2698, 2703, 2714, 2719, 2723, 2724, 2729, 2730, 2736, 2739, 2746, 2749, 2752, 2759, 2768, 2771, 2774, 2780, 2787, 2789, 2792, 2794, 2796, 2800, 2803, 2807, 2813, 2818, 2824, 2825, 2829, 2830, 2831, 2834, 2838, 2841, 2844, 2845, 2847, 2852, 2855, 2856, 2857, 2860, 2862, 2864, 2865, 2869, 2870, 2875, 2876, 2879]
[2022-06-12 01:21:46,708][root][INFO] - Validation results: top k documents hits accuracy [0.23130193905817176, 0.3346260387811634, 0.40277008310249307, 0.4454293628808864, 0.4759002770083102, 0.5066481994459834, 0.5310249307479225, 0.550415512465374, 0.5656509695290859, 0.5806094182825485, 0.5955678670360111, 0.6060941828254848, 0.6135734072022161, 0.6202216066481995, 0.6290858725761773, 0.6376731301939058, 0.6451523545706371, 0.6520775623268698, 0.6565096952908587, 0.6617728531855955, 0.6664819944598338, 0.6711911357340721, 0.675623268698061, 0.6822714681440443, 0.6867036011080332, 0.6916897506925208, 0.6958448753462604, 0.6980609418282548, 0.7011080332409972, 0.7044321329639889, 0.7069252077562327, 0.7091412742382271, 0.7132963988919667, 0.7152354570637119, 0.7177285318559556, 0.7204986149584488, 0.7240997229916898, 0.7257617728531855, 0.7274238227146814, 0.728808864265928, 0.7313019390581718, 0.7340720221606648, 0.7354570637119113, 0.7379501385041551, 0.7401662049861496, 0.7423822714681441, 0.7440443213296399, 0.7454293628808865, 0.7473684210526316, 0.7487534626038781, 0.7518005540166205, 0.753185595567867, 0.7542936288088643, 0.7545706371191135, 0.7559556786703601, 0.7562326869806094, 0.7578947368421053, 0.7587257617728532, 0.7606648199445983, 0.7614958448753463, 0.7623268698060942, 0.7642659279778393, 0.7667590027700831, 0.7675900277008311, 0.7684210526315789, 0.7700831024930748, 0.77202216066482, 0.7725761772853186, 0.7734072022160665, 0.7739612188365651, 0.7745152354570637, 0.775623268698061, 0.7764542936288089, 0.7775623268698061, 0.779224376731302, 0.7806094182825485, 0.7822714681440444, 0.7825484764542936, 0.7836565096952909, 0.7839335180055401, 0.7842105263157895, 0.7850415512465374, 0.7861495844875346, 0.7869806094182825, 0.7878116343490305, 0.7880886426592798, 0.7886426592797784, 0.7900277008310249, 0.7908587257617729, 0.7911357340720222, 0.7914127423822714, 0.7922437673130194, 0.792797783933518, 0.7933518005540167, 0.7936288088642659, 0.7947368421052632, 0.7950138504155124, 0.796398891966759, 0.7966759002770083, 0.7975069252077562]
[2022-06-12 01:21:52,279][root][INFO] - Saved results * scores  to nq_test.result
`

**Comment by jzhoubu** [comment]
I see. That was my mistake and I will fix the readme.

Please use this command to fine-tune instead:
python -m torch.distributed.launch --nproc_per_node=8 train_dense_encoder.py
hydra.run.dir=./experiments/finetune_nq/train
model_file=../../pretrain_hlp/train/dpr_biencoder.best
train_datasets=[nq_train] dev_datasets=[nq_dev]
train=finetune_8xV100
For fine-tuning, use `train=finetune_8xV100` for all your commands (training/embedding/evaluation phrase).


While your results are worse than the zero-shot setting, I guess there are also other problems there. Could you also provide me below information so that I can better help:

  1.  What kind and how many GPU cards do you use for your experiments? (Should use 8 x 32GB cards to reproduce)
  2.  Have you changed any configuration, such as batch size, in the `pretrain_8xV100.yaml` file?
  3.  Can you successfully reproduce the zero-shot retrieval performance?
  4.
  5.

Thanks.





Sincerely,

Jiawei

________________________________
发件人: XingWu_UCAS ***@***.***>
发送时间: 2022年6月12日 19:02
收件人: jzhoubu/HLP ***@***.***>
抄送: Jiawei ZHOU ***@***.***>; Mention ***@***.***>
主题: Re: [jzhoubu/HLP] Process wikidump with WikiExtractor (Issue #2)


One more thing, I finetune the provided ckpt hlp20210726.best<https://drive.google.com/file/d/10vg8-6S4Rnn7wZOxfBmn2C4il0oV0Oyk/view?usp=sharing> on NQ dataset:

python -m torch.distributed.launch --nproc_per_node=8 train_dense_encoder.py
hydra.run.dir=./experiments/finetune_nq/train
model_file=../../pretrain_hlp/train/dpr_biencoder.best
train_datasets=[nq_train] dev_datasets=[nq_dev]
train=pretrain_8xV100

Then I generate the embedding and perform the retrieval evalutaion. But the results are lower than zeroshot, I am confused.

[2022-06-12 01:21:41,611][dpr.data.qa_validation][INFO] - Matching answers in top docs... [2022-06-12 01:21:46,691][dpr.data.qa_validation][INFO] - Per question validation results len=3610 [2022-06-12 01:21:46,708][root][INFO] - Validation results: top k documents hits [835, 1208, 1454, 1608, 1718, 1829, 1917, 1987, 2042, 2096, 2150, 2188, 2215, 2239, 2271, 2302, 2329, 2354, 2370, 2389, 2406, 2423, 2439, 2463, 2479, 2497, 2512, 2520, 2531, 2543, 2552, 2560, 2575, 2582, 2591, 2601, 2614, 2620, 2626, 2631, 2640, 2650, 2655, 2664, 2672, 2680, 2686, 2691, 2698, 2703, 2714, 2719, 2723, 2724, 2729, 2730, 2736, 2739, 2746, 2749, 2752, 2759, 2768, 2771, 2774, 2780, 2787, 2789, 2792, 2794, 2796, 2800, 2803, 2807, 2813, 2818, 2824, 2825, 2829, 2830, 2831, 2834, 2838, 2841, 2844, 2845, 2847, 2852, 2855, 2856, 2857, 2860, 2862, 2864, 2865, 2869, 2870, 2875, 2876, 2879] [2022-06-12 01:21:46,708][root][INFO] - Validation results: top k documents hits accuracy [0.23130193905817176, 0.3346260387811634, 0.40277008310249307, 0.4454293628808864, 0.4759002770083102, 0.5066481994459834, 0.5310249307479225, 0.550415512465374, 0.5656509695290859, 0.5806094182825485, 0.5955678670360111, 0.6060941828254848, 0.6135734072022161, 0.6202216066481995, 0.6290858725761773, 0.6376731301939058, 0.6451523545706371, 0.6520775623268698, 0.6565096952908587, 0.6617728531855955, 0.6664819944598338, 0.6711911357340721, 0.675623268698061, 0.6822714681440443, 0.6867036011080332, 0.6916897506925208, 0.6958448753462604, 0.6980609418282548, 0.7011080332409972, 0.7044321329639889, 0.7069252077562327, 0.7091412742382271, 0.7132963988919667, 0.7152354570637119, 0.7177285318559556, 0.7204986149584488, 0.7240997229916898, 0.7257617728531855, 0.7274238227146814, 0.728808864265928, 0.7313019390581718, 0.7340720221606648, 0.7354570637119113, 0.7379501385041551, 0.7401662049861496, 0.7423822714681441, 0.7440443213296399, 0.7454293628808865, 0.7473684210526316, 0.7487534626038781, 0.7518005540166205, 0.753185595567867, 0.7542936288088643, 0.7545706371191135, 0.7559556786703601, 0.7562326869806094, 0.7578947368421053, 0.7587257617728532, 0.7606648199445983, 0.7614958448753463, 0.7623268698060942, 0.7642659279778393, 0.7667590027700831, 0.7675900277008311, 0.7684210526315789, 0.7700831024930748, 0.77202216066482, 0.7725761772853186, 0.7734072022160665, 0.7739612188365651, 0.7745152354570637, 0.775623268698061, 0.7764542936288089, 0.7775623268698061, 0.779224376731302, 0.7806094182825485, 0.7822714681440444, 0.7825484764542936, 0.7836565096952909, 0.7839335180055401, 0.7842105263157895, 0.7850415512465374, 0.7861495844875346, 0.7869806094182825, 0.7878116343490305, 0.7880886426592798, 0.7886426592797784, 0.7900277008310249, 0.7908587257617729, 0.7911357340720222, 0.7914127423822714, 0.7922437673130194, 0.792797783933518, 0.7933518005540167, 0.7936288088642659, 0.7947368421052632, 0.7950138504155124, 0.796398891966759, 0.7966759002770083, 0.7975069252077562] [2022-06-12 01:21:52,279][root][INFO] - Saved results * scores to nq_test.result

―

**Comment by jzhoubu** [comment]
Additionally, please also make sure your pretrained ckpt file
`../../pretrain_hlp/train/dpr_biencoder.best`
is exactly the provided hlp20210726.best<https://drive.google.com/file/d/10vg8-6S4Rnn7wZOxfBmn2C4il0oV0Oyk/view?usp=sharing> ckpt.




Sincerely,

Jiawei

________________________________
发件人: XingWu_UCAS ***@***.***>
发送时间: 2022年6月12日 19:02
收件人: jzhoubu/HLP ***@***.***>
抄送: Jiawei ZHOU ***@***.***>; Mention ***@***.***>
主题: Re: [jzhoubu/HLP] Process wikidump with WikiExtractor (Issue #2)


One more thing, I finetune the provided ckpt hlp20210726.best<https://drive.google.com/file/d/10vg8-6S4Rnn7wZOxfBmn2C4il0oV0Oyk/view?usp=sharing> on NQ dataset:

python -m torch.distributed.launch --nproc_per_node=8 train_dense_encoder.py
hydra.run.dir=./experiments/finetune_nq/train
model_file=../../pretrain_hlp/train/dpr_biencoder.best
train_datasets=[nq_train] dev_datasets=[nq_dev]
train=pretrain_8xV100

Then I generate the embedding and perform the retrieval evalutaion. But the results are lower than zeroshot, I am confused.

[2022-06-12 01:21:41,611][dpr.data.qa_validation][INFO] - Matching answers in top docs... [2022-06-12 01:21:46,691][dpr.data.qa_validation][INFO] - Per question validation results len=3610 [2022-06-12 01:21:46,708][root][INFO] - Validation results: top k documents hits [835, 1208, 1454, 1608, 1718, 1829, 1917, 1987, 2042, 2096, 2150, 2188, 2215, 2239, 2271, 2302, 2329, 2354, 2370, 2389, 2406, 2423, 2439, 2463, 2479, 2497, 2512, 2520, 2531, 2543, 2552, 2560, 2575, 2582, 2591, 2601, 2614, 2620, 2626, 2631, 2640, 2650, 2655, 2664, 2672, 2680, 2686, 2691, 2698, 2703, 2714, 2719, 2723, 2724, 2729, 2730, 2736, 2739, 2746, 2749, 2752, 2759, 2768, 2771, 2774, 2780, 2787, 2789, 2792, 2794, 2796, 2800, 2803, 2807, 2813, 2818, 2824, 2825, 2829, 2830, 2831, 2834, 2838, 2841, 2844, 2845, 2847, 2852, 2855, 2856, 2857, 2860, 2862, 2864, 2865, 2869, 2870, 2875, 2876, 2879] [2022-06-12 01:21:46,708][root][INFO] - Validation results: top k documents hits accuracy [0.23130193905817176, 0.3346260387811634, 0.40277008310249307, 0.4454293628808864, 0.4759002770083102, 0.5066481994459834, 0.5310249307479225, 0.550415512465374, 0.5656509695290859, 0.5806094182825485, 0.5955678670360111, 0.6060941828254848, 0.6135734072022161, 0.6202216066481995, 0.6290858725761773, 0.6376731301939058, 0.6451523545706371, 0.6520775623268698, 0.6565096952908587, 0.6617728531855955, 0.6664819944598338, 0.6711911357340721, 0.675623268698061, 0.6822714681440443, 0.6867036011080332, 0.6916897506925208, 0.6958448753462604, 0.6980609418282548, 0.7011080332409972, 0.7044321329639889, 0.7069252077562327, 0.7091412742382271, 0.7132963988919667, 0.7152354570637119, 0.7177285318559556, 0.7204986149584488, 0.7240997229916898, 0.7257617728531855, 0.7274238227146814, 0.728808864265928, 0.7313019390581718, 0.7340720221606648, 0.7354570637119113, 0.7379501385041551, 0.7401662049861496, 0.7423822714681441, 0.7440443213296399, 0.7454293628808865, 0.7473684210526316, 0.7487534626038781, 0.7518005540166205, 0.753185595567867, 0.7542936288088643, 0.7545706371191135, 0.7559556786703601, 0.7562326869806094, 0.7578947368421053, 0.7587257617728532, 0.7606648199445983, 0.7614958448753463, 0.7623268698060942, 0.7642659279778393, 0.7667590027700831, 0.7675900277008311, 0.7684210526315789, 0.7700831024930748, 0.77202216066482, 0.7725761772853186, 0.7734072022160665, 0.7739612188365651, 0.7745152354570637, 0.775623268698061, 0.7764542936288089, 0.7775623268698061, 0.779224376731302, 0.7806094182825485, 0.7822714681440444, 0.7825484764542936, 0.7836565096952909, 0.7839335180055401, 0.7842105263157895, 0.7850415512465374, 0.7861495844875346, 0.7869806094182825, 0.7878116343490305, 0.7880886426592798, 0.7886426592797784, 0.7900277008310249, 0.7908587257617729, 0.7911357340720222, 0.7914127423822714, 0.7922437673130194, 0.792797783933518, 0.7933518005540167, 0.7936288088642659, 0.7947368421052632, 0.7950138504155124, 0.796398891966759, 0.7966759002770083, 0.7975069252077562] [2022-06-12 01:21:52,279][root][INFO] - Saved results * scores to nq_test.result

―

**Comment by jzhoubu** [comment]
@1024er Not sure if that email can reach you. For your fine-tuning question, I will also reply here.

Please use this command to fine-tune instead:
```
python -m torch.distributed.launch --nproc_per_node=8 train_dense_encoder.py
hydra.run.dir=./experiments/finetune_nq/train
model_file=../../pretrain_hlp/train/dpr_biencoder.best
train_datasets=[nq_train] dev_datasets=[nq_dev]
train=finetune_8xV100
```
For fine-tuning, use `train=finetune_8xV100` for all the commands (training/embedding/evaluation phrase).


While your results are worse than the zero-shot setting, I guess main be also other problems there. Could you also provide me below information so that I can better help:

  1.  What kind and how many GPU cards do you use for your experiments? (Should use 8 x 32GB cards to reproduce)
  2.  Have you changed any configuration, such as batch size, in the `pretrain_8xV100.yaml` file?
  3.  Can you successfully reproduce the zero-shot retrieval performance?
  4.  Is the ckpt file `../../pretrain_hlp/train/dpr_biencoder.best` the same one to our provided ckpt [hlp20210726.best](https://drive.google.com/file/d/10vg8-6S4Rnn7wZOxfBmn2C4il0oV0Oyk/view?usp=sharing)?

Thanks.

**Comment by 1024er** [comment]
Finally, with the help of @jzhoubu , I reproduce the results  on NQ dataset.  Thank you so much !

|  | NQ |
|:-----:|:----: |
| Zero-shot<br>Fine-tune | top5 / top20 / top100 |
| **HLP (origin)** | 51.2 / 70.2 / 82.0<br>70.9 / 81.4 / 88.0 |
| **HLP (latest)** | 50.9 / 69.3 / 82.1<br>70.6 / 81.3 / 88.0 |
| **reproduce (on 8xA100)** | 51.6 / 69.1 / 82.2<br>70.2 / 81.6 / 87.8 |
| **reproduce (80% data on 8xV100)** | 50.6 / 69.4 / 82.2<br>69.1 / 81.1 / 87.4 |
